Android/app/src/main/java/io/github/project_travel_mate/notifications/NotificationsAdapter.java

```diff
@@ -44,11 +72,62 @@ public View getView(final int position, View view, @NonNull ViewGroup parent) {
         holder.name.setText(mNotifications.get(position).getText());
         if (mNotifications.get(position).isRead()) {
             holder.readStatus.setVisibility(View.INVISIBLE);
+        } else {
+            holder.readStatus.setVisibility(View.VISIBLE);
         }
-        view.setOnClickListener(view1 -> mNotifications.get(position).setRead(true));
         view.setFocusable(false);
+        view.setOnClickListener(view12 -> {
+            if (!mNotifications.get(position).isRead()) {
+                markAsRead(position);
+            }
+            openTripPage(position);
+        });
         return view;
     }
+    public void markAsRead(final int position) {
+        mDialog = new MaterialDialog.Builder(mContext)
```

Comment:
Don't show the loader of marking the notification as read. Let it be in the background only.